src/main/scala/exercises/06_Collections.scala

```diff
@@ -113,9 +99,7 @@ object Collections {
    * As always, use either build-in functions, or roll your own way via a
    * pattern match or some other method.
    */
-  def oddElements(iList: List[Int]): List[Int] = {
-    error("fix me")
-  }
+  def oddElements(iList: List[Int]): List[Int] = iList.foldRight(List(): List[Int])({(l,acc) => if (l % 2 != 0)  l::acc else acc })
```

Comment:
I agree that filter would have been cleaner here!

As far as choosing the fold I used, it is more efficient to cons to the front of a list than it is to append to the end, therefore `foldRight` is better than `foldLeft` in this situation. `fold` along is non-deterministic in ordering, so that was not suitable.

Here is functionally equivalent, but less optimal code using a `foldLeft`:

```scala
def oddElementpp(iList: List[Int]): List[Int] = iList.foldLeft(List(): List[Int])({(acc,l) => if (l % 2 != 0) acc ++ List(l) else acc })
```

The arguments of the annon. function are flipped and we append the next element to the accumulator as a singleton list.